### PR TITLE
feat: add abac rules for products, environments, and promotion flows

### PR DESCRIPTION
**Pull request by danielm-codefresh**
## Checklist

* [ ] _I have read [CONTRIBUTING.md](https://github.com/codefresh-io/terraform-provider-codefresh/blob/master/CONTRIBUTING.md)._
* [ ] _I have [allowed changes to my fork to be made](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)._
* [ ] _I have added tests, assuming new tests are warranted_.
* [ ] _I understand that the `/test` comment will be ignored by the CI trigger [unless it is made by a repo admin or collaborator](https://codefresh.io/docs/docs/pipelines/triggers/git-triggers/#support-for-building-pull-requests-from-forks)._

### CODE REVIEW

File: codefresh/resource_abac_rules.go [hunk 1]
```diff
@@ -67,6 +70,9 @@ Action to be allowed. Possible values:
 	* TERMINATE_SYNC
 	* VIEW_POD_LOGS
 	* APP_ROLLBACK
+	* "TRIGGER_PROMOTION"
```
Comment (denis-codefresh) [inline codefresh/resource_abac_rules.go:73]: double quotes not needed

File: docs/resources/abac_rules.md [hunk 2]
```diff
@@ -48,8 +48,14 @@ resource "codefresh_abac_rules" "app_rule" {
 	* TERMINATE_SYNC
 	* VIEW_POD_LOGS
 	* APP_ROLLBACK
+	* "TRIGGER_PROMOTION"
```
Comment (denis-codefresh) [inline docs/resources/abac_rules.md:51]: double quotes not needed

**Review by denis-codefresh — APPROVED** [review]

**Review by ilia-medvedev-codefresh — APPROVED** [review]

**Pull request merged**